### PR TITLE
refactor: separate crawler related code to core package

### PR DESCRIPTION
**Pull request by B4nan**

### CODE REVIEW

File: jest.config.js [hunk 1]
```diff
@@ -13,14 +13,20 @@ module.exports = {
...
+        '^@crawlers/(.*)$': '<rootDir>/packages/$1/src',
     },
     modulePathIgnorePatterns: [
         'dist/package.json',
         '<rootDir>/package.json',
     ],
     globals: {
         'ts-jest': {
-            tsconfig: 'test/tsconfig.json',
+            tsconfig: 'test/apify/tsconfig.json',
```
Comment (vladfrangu) [inline jest.config.js:29]: We should probably move the tsconfig to the test folder instead

File: packages/apify/src/actor.ts [hunk 2]
```diff
@@ -1,458 +1,1320 @@
...
+        if (!this.isAtHome()) {
+            log.warning('Actor.metamorph() is only supported when running on the Apify platform.');
+            return;
+        }
+
+        const {
+            customAfterSleepMillis = this.config.get('metamorphAfterSleepMillis'),
+            ...metamorphOpts
+        } = options;
+        const runId = this.config.get('actorRunId')!;
```
Comment (vladfrangu) [inline packages/apify/src/actor.ts:301]: Probably a nit for the future but we should probably have an object/enum for these keys, so we can strictly type them

File: packages/apify/src/actor.ts [hunk 3]
```diff
@@ -1,458 +1,1320 @@
...
+        await this.newClient().run(runId).metamorph(targetActorId, input, metamorphOpts);
```
Comment (vladfrangu) [inline packages/apify/src/actor.ts:302]: Shouldn't you reuse this.apifyClient instead?

File: packages/apify/src/actor.ts [hunk 4]
```diff
@@ -1,458 +1,1320 @@
...
+        }));
+
+        const { eventTypes, requestUrl, payloadTemplate, idempotencyKey } = options;
+
+        if (!this.isAtHome()) {
+            log.warning('Actor.addWebhook() is only supported when running on the Apify platform. The webhook will not be invoked.');
+            return undefined;
+        }
+
+        const runId = process.env[ENV_VARS.ACTOR_RUN_ID];
```
Comment (vladfrangu) [inline packages/apify/src/actor.ts:334]: Shouldn't this use this.config.get, for consistency?

**Review by vladfrangu — CHANGES_REQUESTED** [review]

File: packages/apify/src/actor.ts (hunk 2 above)
Comment (B4nan) [reply packages/apify/src/actor.ts:301]: what you mean? config is already strictly typed, it both checks the keys as well as maps the return type based on it

File: packages/apify/src/actor.ts (hunk 3 above)
Comment (B4nan) [reply packages/apify/src/actor.ts:302]: i believe there was some reason originally, but now its all rewritten so it should preserve the old behaviour from static helpers

so yes, i guess it should be safe to use it here (tests do pass)

**Review by vladfrangu — APPROVED** [review]

**Pull request merged**